File: leetCode/fukun/001-100/lc_061.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}
const rotateRight = (head, k) => {
    if (!head) return head;
    let curr = head,
        tmp = null,
        n = 0;

    // 计算长度并接续尾部
    while (1) {
        n++;
        if (!curr.next) {
            curr.next = head;
            break;
        }
        curr = curr.next;
    }

    //找到头部断开头部
    k = k % n;
    while (k++ < n) {
        if(k === n) tmp = head;
        head = head.next;
    }
    tmp.next = null;
    return head;
};

let ln0 = new ListNode(0);
let ln1 = new ListNode(1);
let ln2 = new ListNode(2);
ln0.next = ln1;
ln1.next = ln2;
console.log(rotateRight(ln0, 4));

File: leetCode/lwc/0001-0100/lc_061.java
package solution2;

/**
 * @author lwc
 * @date 2019/10/11 16:49
 */
public class U61 {
    public ListNode rotateRight(ListNode head, int k) {
        if(head == null) return null;
        ListNode l1 = head;
        ListNode l2 = head;
        int v = 1;

        while(l1.next != null && ++v > 0)
            l1 = l1.next;
        int pos = v - (k % v);

        for(v = 1; v < pos; v++)
            l2 = l2.next;

        l1.next = head;
        l1 = l2.next;
        l2 = l2.next = null;

        return l1;
    }
}
